modules/ROOT/pages/sync/auth.adoc

```diff
@@ -0,0 +1,49 @@
+include::{partialsdir}/attributes.adoc[]
+
+//':context:' is a vital parameter. See: http://asciidoctor.org/docs/user-manual/#include-multiple
+:context: sync_auth
+
+[id='{context}_authentication']
+= Authentication
+
+By default, {sync-service} does not provide any authentication and authorization mechanism.
+Users can configure {sync-service} to work with {keycloak-service} to provide authentication and 
+authorization for their applications.
+
+== Prerequisites
+
+* Ensure {sync-service} is xref:sync/provisioning.adoc[provisioned].
+* Ensure {keycloak-service} is xref:keycloak/provisioning.adoc[provisioned]
+* Ensure {keycloak-service} is xref:keycloak/coding.adoc[configured] to work with your mobile applications
+
+== Binding the {sync-service} service to a {keycloak-service} Instance
+
+To benefit from authentication, developers need to connect (bind) {sync-service} into  {keycloak-service} Instance. 
+
+== Creating the binding secret from a Keycloak Instance
+
+* Navigate to your Openshift project
+* Select *Applications* and then *Provisioned Services*
+* Copy the *Instance Name* field from the *{sync-service}* service
+* Select the *Identity Management* service from the list of Provisioned Services
+* Select the *Create Binding* option
+* Enter the copied *Instance Name* of the provisioned *{sync-service}* service
+
+A secret will be created as a result of this binding.
+
+== Making the Secret available to the {sync-service} 
+
+* Select *View Secret* on the *Identity Management* details page
+* Select *Add to Application* and select the data-sync-server from the options
+* Select the *Volume* option and enter the following mount path: `/opt/keycloak`
+
+NOTE: The mount path must be set to `/opt/keycloak` for the {sync-service} to be configured to use the secret.
+
+== Making Secret available to application
+
+By default, the secret is not visible to the server application.
+o point the server to the application, please add a new environment variable:
+
+----
+KEYCLOAK_CONFIG_FILE= `/opt/keycloak`TODO
```

Comment:
are we leaving this as is? we could change it to the actual value we are using for now?

Reply:
Ups. yeah. TODO needs to be removed. I overlooked that in the rush